Remove NVDA service, no longer needed as NVDA will require Windows 7 SP1 or higher

josephsl opened the pull request:
### Link to issue number:
None

### Summary of the issue:
Removes NVDA service which was needed in Windows XP, now that Windows 7 SP1 is the minimum requirement for NVDA.

### Description of how this pull request fixes the issue:
This pull request modifies setup and installer to:

1. Tells py2exe to not build the service.
2. Do not install NVDA service nor start it.
3. Service module is removed.

### Testing performed:
Successfully compiled NVDA without the service executable.

### Known issues with pull request:
None

### Change log entry:
None

josephsl committed:
NVDA service: py2exe will no longer build the service module, as the service is only needed on XP and Server 2003.

josephsl committed:
Installer: no longer need to start NVDA service, as Windows 7 SP1 or higher is required now.

josephsl committed:
NVDA service is hereby removed.

leonardder commented on source/installer.py:
```diff
@@ -192,10 +192,6 @@ def registerInstallation(installDir,startMenuFolder,shouldCreateDesktopShortcut,
 			_winreg.SetValueEx(k,config.CONFIG_IN_LOCAL_APPDATA_SUBKEY,None,_winreg.REG_DWORD,int(configInLocalAppData))
 	if easeOfAccess.isSupported:
```
This makes no sense any more, as this will always be True.

josephsl commented:
Hi, that one and several commits that’ll modernize the API (including always making NVDA high DPI aware) is headed for a separate PR, as it is better to do one thing at a time. Thanks.

 

From: Leonard de Ruijter [mailto:notifications@github.com] 
Sent: Sunday, September 24, 2017 11:52 PM
To: nvaccess/nvda <nvda@noreply.github.com>
Cc: Joseph Lee <joseph.lee22590@gmail.com>; Assign <assign@noreply.github.com>
Subject: Re: [nvaccess/nvda] Remove NVDA service, no longer requird as NVDA will require Windows 7 SP1 or higher (#7625)

 

@leonardder commented on this pull request.

  _____  

In source/installer.py <https://github.com/nvaccess/nvda/pull/7625#discussion_r140699535> :

> @@ -192,10 +192,6 @@ def registerInstallation(installDir,startMenuFolder,shouldCreateDesktopShortcut,
                _winreg.SetValueEx(k,config.CONFIG_IN_LOCAL_APPDATA_SUBKEY,None,_winreg.REG_DWORD,int(configInLocalAppData))
        if easeOfAccess.isSupported:

This makes no sense any more, as this will always be True.

—
You are receiving this because you were assigned.
Reply to this email directly, view it on GitHub <https://github.com/nvaccess/nvda/pull/7625#pullrequestreview-64806967> , or mute the thread <https://github.com/notifications/unsubscribe-auth/AHgLkJHhxT_eTrGNxok0l8HKjcgPTbtBks5sl02ZgaJpZM4PiP85> .

josephsl committed:
Just assume Ease of Access is available.

Noted by Leonard de Ruijter: just asusme Ease of Access is present.

josephsl committed:
NVDA slave: proper copyright header, no longer manages NVDA service.

Added proper copyright header for the slave process. Also, as XP is no longer supported, service launcher is no more.

josephsl committed:
Config: simplify secure screen startup by checking if NVDA is installed.

No more support for XP/Vista/7 RTM means:
* No more NVDA service.
* No more check for presence of Ease of Access.
Thus, the secure screen startup checker just returns if NVDA is installed, thus ready to be transformed into a flag in a future release.

michaelDCurran commented:
A reference to nvda_service.exe must be removed from sconstruct. This is causing signed builds to fail.

michaelDCurran reviewed (changes requested):
nvda_service.exe must be removed from sconstruct

josephsl committed:
Merge branch 'master' into removeNVDAService

josephsl committed:
SConstruct: no more service builder.

Reviewed by Mick Curran (NV Access): knew something was incomplete when working on removing NVDA service - SCons should no longer be told to build the service executable, thereby allowing signed builds to proceed.

michaelDCurran commented:
add-ons such as goldenCursor and windows7Mangnifier depend on win32api, which is no longer included as it was an implicit dependency on nvda_service.
It would be great if these add-ons could be rewritten to not use win32api, but for this next release at least, we should manually include win32api in NVDA so these add-ons do not break.
Refer to how we explicitly include the bisect module in setup.py.

josephsl committed:
Merge branch 'master' into removeNVDAService

josephsl committed:
NvDA service: include win32api so some add-ons can function properly.

The following add-on is known for using win32api: Golden Cursor. Because of the impact of this add-on, include win32api in 2017.4, to be removed in 2018 as part of Python 3 transition or a major code rewrite.

derekriemer commented on source/setup.py:
```diff
@@ -205,8 +205,9 @@ def getRecursiveDataFiles(dest,source,excludes=()):
-		# Explicitly include it so we don't break some add-ons.
-		"includes": ["nvdaBuiltin", "bisect"],
+		# Also, the service executable used win32api, which some add-ons use for various purposes.
+		# Explicitly include them so we don't break some add-ons.
```
nit: them, -> it.
Also, what about win32con for a while?

josephsl commented:
Hi, I need to find out if other add-ons use that module.

 

From: Derek Riemer [mailto:notifications@github.com] 
Sent: Tuesday, October 17, 2017 11:01 AM
To: nvaccess/nvda <nvda@noreply.github.com>
Cc: Joseph Lee <joseph.lee22590@gmail.com>; Assign <assign@noreply.github.com>
Subject: Re: [nvaccess/nvda] Remove NVDA service, no longer needed as NVDA will require Windows 7 SP1 or higher (#7625)

 

@derekriemer commented on this pull request.

  _____  

In source/setup.py <https://github.com/nvaccess/nvda/pull/7625#discussion_r145204445> :

> @@ -205,8 +205,9 @@ def getRecursiveDataFiles(dest,source,excludes=()):
                       "serial.loopback_connection", "serial.rfc2217", "serial.serialcli", "serial.serialjava", "serial.serialposix", "serial.socket_connection"],
               "packages": ["NVDAObjects","virtualBuffers","appModules","comInterfaces","brailleDisplayDrivers","synthDrivers"],
               # #3368: bisect was implicitly included with Python 2.7.3, but isn't with 2.7.5.
-              # Explicitly include it so we don't break some add-ons.
-              "includes": ["nvdaBuiltin", "bisect"],
+              # Also, the service executable used win32api, which some add-ons use for various purposes.
+              # Explicitly include them so we don't break some add-ons.

nit: them, -> it.
Also, what about win32con for a while?

—
You are receiving this because you were assigned.
Reply to this email directly, view it on GitHub <https://github.com/nvaccess/nvda/pull/7625#pullrequestreview-69970179> , or mute the thread <https://github.com/notifications/unsubscribe-auth/AHgLkFI1xdySZWoFgMO0sjU7PCOrc_49ks5stOtJgaJpZM4PiP85> .

Brian1Gaff commented:
Is this the point though? Is it not that its a good file to keep if its only 
small as a resource unless it cocks up something that will change later on 
when you move versions?

Brian

bglists@blueyonder.co.uk
Sent via blueyonder.
Please address personal email to:-
briang1@blueyonder.co.uk, putting 'Brian Gaff'
in the display name field.
-----

Brian1Gaff commented:
What worries me a little here is that do we know if any already written app 
modules in nvda use these files?
If they do then some software, although old may suddenly stop working.

Brian

bglists@blueyonder.co.uk
Sent via blueyonder.
Please address personal email to:-
briang1@blueyonder.co.uk, putting 'Brian Gaff'
in the display name field.
----- Original Message ----- 
From: "Joseph Lee" <notifications@github.com>
To: "nvaccess/nvda" <nvda@noreply.github.com>
Cc: "Subscribed" <subscribed@noreply.github.com>
Sent: Tuesday, October 17, 2017 8:15 PM
Subject: Re: [nvaccess/nvda] Remove NVDA service, no longer needed as NVDA 
will require Windows 7 SP1 or higher (#7625)


Hi, I need to find out if other add-ons use that module.



From: Derek Riemer [mailto:notifications@github.com]
Sent: Tuesday, October 17, 2017 11:01 AM
To: nvaccess/nvda <nvda@noreply.github.com>
Cc: Joseph Lee <joseph.lee22590@gmail.com>; Assign 
<assign@noreply.github.com>
Subject: Re: [nvaccess/nvda] Remove NVDA service, no longer needed as NVDA 
will require Windows 7 SP1 or higher (#7625)



@derekriemer commented on this pull request.

  _____

In source/setup.py 
<https://github.com/nvaccess/nvda/pull/7625#discussion_r145204445> :

> @@ -205,8 +205,9 @@ def getRecursiveDataFiles(dest,source,excludes=()):
                       "serial.loopback_connection", "serial.rfc2217", 
"serial.serialcli", "serial.serialjava", "serial.serialposix", 
"serial.socket_connection"],
               "packages": 
["NVDAObjects","virtualBuffers","appModules","comInterfaces","brailleDisplayDrivers","synthDrivers"],
               # #3368: bisect was implicitly included with Python 2.7.3, 
but isn't with 2.7.5.
-              # Explicitly include it so we don't break some add-ons.
-              "includes": ["nvdaBuiltin", "bisect"],
+              # Also, the service executable used win32api, which some 
add-ons use for various purposes.
+              # Explicitly include them so we don't break some add-ons.

nit: them, -> it.
Also, what about win32con for a while?

—
You are receiving this because you were assigned.
Reply to this email directly, view it on GitHub 
<https://github.com/nvaccess/nvda/pull/7625#pullrequestreview-69970179> , or 
mute the thread 
<https://github.com/notifications/unsubscribe-auth/AHgLkFI1xdySZWoFgMO0sjU7PCOrc_49ks5stOtJgaJpZM4PiP85> 
.



-- 
You are receiving this because you are subscribed to this thread.
Reply to this email directly or view it on GitHub:
https://github.com/nvaccess/nvda/pull/7625#issuecomment-337339404

derekriemer commented:
we can (and must) fix those. They're in core.

Brian1Gaff commented:
I wonder if this removal was a little premature, and what is needed is a way 
to include the bits that are always needed in the  main nvda system, and 
then only remove the rest of the service, unless people have taken 
advantage of it being a service and hence need to have the items available 
in that way of course.
 Apart from breaking more ties with XP was there really any point in 
removing the service in the first place?
 Brian

bglists@blueyonder.co.uk
Sent via blueyonder.
Please address personal email to:-
briang1@blueyonder.co.uk, putting 'Brian Gaff'
in the display name field.
-----
